src/ui-kit/form-controls/date/date.component.ts

```diff
@@ -490,19 +499,19 @@ export class SamDateComponent
   }
 
   triggerTouch() {
-    this.onTouched();
+    // this.onTouched();
   }
 
   triggerMonthTouch(event) {
-    if (event.target.value.substring(0, 1) === '0') {
-      this.month.nativeElement.value = event.target.value.substring(1);
-    }
+    // if (event.target.value.substring(0, 1) === '0') {
+    //   this.month.nativeElement.value = event.target.value.substring(1);
+    // }
     this.onTouched();
   }
   triggerDayTouch(event) {
-    if (event.target.value.substring(0, 1) === '0') {
```

Comment:
@yerramshilpa Same comment here -- for all these triggerTouch, since they are public, we are limited in what we can do with them. This is going to be a breaking change for us.